### DISCUSSION TITLE
Compression + Encryption question

### DISCUSSION THREAD

**Discussion by tsohst** [minio/minio#16307] [Q&A] [answered]
Hello.

I found that compression is not enabled by default.
While reading the documentation, I saw that ```Combining encryption and compression is not safe in all setups.```

Currently we have a KMS configured and encryption is working.
```
0 minio2 ~ # mc admin kms key status stage
Key: stage-encryption-key
   - Encryption ✔
   - Decryption ✔
```

But when asking the compression config, it sais encryption is not enabled.
```
0 minio2 ~ # mc admin config get stage compression
compression enable=off allow_encryption=off extensions=.txt,.log,.csv,.json,.tar,.xml,.bin mime_types=text/*,application/json,application/xml,binary/octet-stream
```

Do both encryption options correlate in some way?
Can we enable compression in combination with a KMS?

https://github.com/minio/minio/blob/master/docs/compression/README.md

**Comment by klauspost** [comment, marked as answer]
Please see this blog post: https://blog.min.io/transparent-data-compression/

That should cover all your questions.
